### DISCUSSION TITLE
Tailscale - Adguard DNS rewrite

### DISCUSSION THREAD

**Discussion by Sithuk** [AdguardTeam/AdGuardHome#7263] [General]
I have successfully setup tailscale to connect to my local network and use my AdGuardHome instance for DNS resolution. However, the shortcuts I have defined in DNS Rewrites and Custom Filter Rules do not work. I can use the 192.168.x.x address to connect to specific machine without a problem, but the shortcut isn't recognised (e.g. shortcut.lan).

The shortcuts work fine when I am on a machine in my local network. Just not when using Tailscale to connect remotely.

I’ve tried adding the Tailscale ipv4 range to the AdGuard conf file private_networks section and installing Tailscale on the adguardhome machine. I made sure to restart AdguardHome so the conf file was loaded anew.

I’ve also tried adding the tailnet address for the adguard machine as the Tailnet nameserver instead of the 192.168.x.x address, but none of that caused the shortcuts to work.

Does anyone know how to get shortcuts to work with Tailscale?

**Comment by guipace** [comment]
+1 on this; I have the same issue

**Comment by DataHearth** [comment]
Hey! I managed to do it with my tailnet. What I did:
- When adding the `IP` in the rewrite section for a specific `DN`, you should set the IP to the machine IN the tailnet. `e.g: machine1 192.168.1.x` became => `machine1 100.97.x.x` for example. 
- Quick tip, depending on the `TTL` set for a given `DN`, after changing my `IP`s in rewrite section, I flush my devices DNS cache. New query should now return the tailnet `IP` and not hit the cache right away.

Option two I'm exploring right now: 
- Setting a subnet router in your tailnet. The benefit of this, when in your local network, no need to enable the tailscale client to route your traffic and get a valid `DN` record. Downside, I don't how to configure everything yet and I haven't tested this.

EDIT: Quick note for solution one, I've set the option to get my DNS settings from tailscale server. In my tailnet, I've set the first two entries to my tailnet machine IPv4/6. You must then set your clients' configuration to get your DNS from the tailscale server (if not set manually by your OS)
